app/wizards/schools/register_ect_wizard/ect.rb

```diff
@@ -98,6 +101,80 @@ def school_led?
       def trs_full_name
         Teachers::Name.new(self).full_name_in_trs
       end
+
+      def previously_registered?
+        previous_ect_at_school_period.present?
+      end
+
+      def induction_start_date
+        first_induction_period&.started_on
+      end
+
+      def previous_appropriate_body_name
+        previous_appropriate_body&.name
+      end
+
+      def previous_training_programme
+        previous_ect_at_school_period&.programme_type
+      end
+
+      def previous_provider_led?
+        previous_ect_at_school_period&.provider_led_programme_type?
+      end
+
+      def previous_lead_provider_name
+        previous_lead_provider&.name
+      end
+
+      def previous_delivery_partner_name
+        previous_delivery_partner&.name
+      end
+
+      def previous_school
+        previous_ect_at_school_period&.school
+      end
+
+    private
+
+      def previous_training_period
+        @previous_training_period ||= TrainingPeriods::Search
+          .new(order: :started_on)
+          .training_periods(ect_id: previous_ect_at_school_period.id)
+          .last
+      end
+
+      def previous_lead_provider
+        previous_training_period&.lead_provider
+      end
+
+      def previous_delivery_partner
+        previous_training_period&.delivery_partner
+      end
+
+      def previous_ect_at_school_period
+        @previous_ect_at_school_period ||= ECTAtSchoolPeriods::Search
+          .new(order: :started_on)
+          .ect_periods(trn:)
+          .last
+      end
+
+      def previous_induction_period
+        ordered_induction_periods.last
+      end
+
+      def first_induction_period
+        ordered_induction_periods.first
+      end
+
+      def ordered_induction_periods
+        @ordered_induction_periods ||= InductionPeriods::Search
+          .new(order: :started_on)
+          .induction_periods(trn:)
+      end
+
+      def previous_appropriate_body
+        previous_induction_period&.appropriate_body
+      end
```

Comment:
Looks nice to me! 
Can we keep the alphabetic order of the methods in the class so it is easier to read?

Reply:
Sure!